src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorManagedRuleObject.cs

```diff
@@ -31,15 +26,15 @@ namespace Microsoft.Azure.Commands.FrontDoor.Cmdlets
     public class NewAzureRmFrontDoorManagedRuleObject : AzureFrontDoorCmdletBase
     {
         /// <summary>
-        /// Describes priority of the rule
+        /// Type of the ruleset (e.g.: DefaultRuleSet)
         /// </summary>
-        [Parameter(Mandatory = true, HelpMessage = "Describes priority of the rule")]
-        public int Priority { get; set; }
+        [Parameter(Mandatory = true, HelpMessage = "Type of the ruleset")]
+        public string Type { get; set; }
```

Comment:
Is there a know set of types?  In this case, this should have a completer

Reply:
There is but we might add some later and we would not like to change the cmdlets whenever we add another.

Reply:
You need to add a completer for the known set of types - this is essentially the same as an expandable enum.  See: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/azure-powershell-developer-guide.md#generic-argument-completer